src/index.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  CreateQueueCommand,
+  GetQueueAttributesCommand,
+  SetQueueAttributesCommand,
+  SQSClient
+} from '@aws-sdk/client-sqs'
+
+import {
+  SetSubscriptionAttributesCommand,
+  SNSClient,
+  SubscribeCommand
+} from '@aws-sdk/client-sns'
+
+import type {
+  CreateQueueCommandInput,
+  GetQueueAttributesCommandInput,
+  GetQueueAttributesCommandOutput,
+  SetQueueAttributesCommandInput,
+  CreateQueueCommandOutput,
+  SetQueueAttributesCommandOutput
+} from '@aws-sdk/client-sqs'
+
+import type {
+  SetSubscriptionAttributesCommandInput,
+  SubscribeCommandInput,
+  SubscribeCommandOutput,
+  SetSubscriptionAttributesCommandOutput
+} from '@aws-sdk/client-sns'
+
+import {
+  CreateSqsQueueParams,
+  SqnsOptions,
+  TopicOptions
+} from './types'
+
+const sqns = async (options: SqnsOptions = {}): Promise<string> => {
+  const {
+    region,
+    queueName,
+    maxReceiveCount = 3,
+    topic = {}
+  } = options
+
+  if (region === null || region === undefined || region === '') throw Error('Missing region')
+  if (queueName === null || queueName === undefined || queueName === '') throw Error('Missing queueName')
+
+  const topicOptions: TopicOptions = {
+    rawMessageDelivery: true,
+    ...topic
+  }
+
+  const sqsClient = new SQSClient({ region })
+  const snsClient = new SNSClient({ region })
+
+  const createQueue = async (params: CreateQueueCommandInput): Promise<CreateQueueCommandOutput> =>
+    await sqsClient.send(new CreateQueueCommand(params))
+
+  const getQueueAttributes = async (params: GetQueueAttributesCommandInput): Promise<GetQueueAttributesCommandOutput> =>
+    await sqsClient.send(new GetQueueAttributesCommand(params))
+
+  const setQueueAttributes = async (params: SetQueueAttributesCommandInput): Promise<SetQueueAttributesCommandOutput> =>
+    await sqsClient.send(new SetQueueAttributesCommand(params))
+
+  const subscribe = async (params: SubscribeCommandInput): Promise<SubscribeCommandOutput> =>
+    await snsClient.send(new SubscribeCommand(params))
+
+  const setSubscriptionAttributes = async (params: SetSubscriptionAttributesCommandInput): Promise<SetSubscriptionAttributesCommandOutput> =>
+    await snsClient.send(new SetSubscriptionAttributesCommand(params))
+
+  const createDeadletterQueue = async (queueName: string): Promise<string | undefined> =>
+    await createQueue({ QueueName: `${queueName}-DLQ` })
+      .then(queue => queue.QueueUrl)
+      .catch(() => undefined)
+
+  const getQueueArn = async (QueueUrl: string): Promise<string | undefined> =>
+    await getQueueAttributes({
+      QueueUrl,
+      AttributeNames: ['QueueArn']
+    })
+      .then((attributes: GetQueueAttributesCommandOutput) => attributes.Attributes?.QueueArn)
```

Comment:
Could we add one more test case covering the scenario when this attribute is not returned? I ran a quick coverage on this branch with your changes and this is the only bit not covered. If we add a test validating this we'll get 100% coverage 🥳 .

![image](https://user-images.githubusercontent.com/13476437/229131878-34f162c5-c77e-43f4-964d-1a9afacd453b.png)


```suggestion
      .then((attributes) => attributes.Attributes?.QueueArn)
```